File: src/longest-streak/solution/longest-streak.ts
export const findLongestStreak = (arr: boolean[], n: number): number => {
	if (arr.length < 1 || arr.length < n) {
		return 0;
	}

	let peak = 1;
	let streak = 1;
	let uut = arr[0];

	for (let i = 1; i < arr.length; i++) {
		const v = arr[i];

		if (v === uut) {
			streak++;
			peak = Math.max(streak, peak);
		} else {
			streak = 1;
			uut = v;
		}
	}

	return peak < n ? 0 : peak;
};

File: GoLang/longest-streak/solution/longest-streak.go
package streak

func LongestStreak(arr []bool, n int) int {
	if len(arr) < n || len(arr) < 1 {
		return 0
	}

	uut := arr[0]
	streak := 1
	peak := 1

	for i := 1; i < len(arr); i++ {
		v := arr[i]

		if uut == v {
			streak++
			peak = max(peak, streak)
		} else {
			streak = 1
			uut = v
		}
	}

	if peak >= n {
		return peak
	}
	return 0
}
